internal/resource/obclusteroperation/obclusteroperation_task.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright (c) 2023 OceanBase
+ob-operator is licensed under Mulan PSL v2.
+You can use this software according to the terms and conditions of the Mulan PSL v2.
+You may obtain a copy of Mulan PSL v2 at:
+         http://license.coscl.org.cn/MulanPSL2
+THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+See the Mulan PSL v2 for more details.
+*/
+
+package obclusteroperation
+
+import (
+	"errors"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oceanbase/ob-operator/api/constants"
+	apitypes "github.com/oceanbase/ob-operator/api/types"
+	v1alpha1 "github.com/oceanbase/ob-operator/api/v1alpha1"
+	obcfg "github.com/oceanbase/ob-operator/internal/config/operator"
+	oceanbaseconst "github.com/oceanbase/ob-operator/internal/const/oceanbase"
+	clusterstatus "github.com/oceanbase/ob-operator/internal/const/status/obcluster"
+	serverstatus "github.com/oceanbase/ob-operator/internal/const/status/observer"
+	"github.com/oceanbase/ob-operator/pkg/task/builder"
+	tasktypes "github.com/oceanbase/ob-operator/pkg/task/types"
+)
+
+//go:generate task_register $GOFILE
+
+var taskMap = builder.NewTaskHub[*OBClusterOperationManager]()
+
+func ModifyClusterSpec(m *OBClusterOperationManager) tasktypes.TaskError {
+	obcluster := &v1alpha1.OBCluster{}
+	err := m.Client.Get(m.Ctx, types.NamespacedName{
+		Namespace: m.Resource.Namespace,
+		Name:      m.Resource.Spec.OBCluster,
+	}, obcluster)
+	if err != nil {
+		m.Logger.Error(err, "Failed to find obcluster")
+		return err
+	}
+	origin := obcluster.DeepCopy()
+	switch strings.ToLower(string(m.Resource.Spec.Type)) {
+	case strings.ToLower(string(constants.ClusterOpTypeAddZones)):
+		if len(m.Resource.Spec.AddZones) == 0 {
+			return errors.New("AddZones is empty")
+		}
+		obcluster.Spec.Topology = append(obcluster.Spec.Topology, m.Resource.Spec.AddZones...)
```

Comment:
check the zone names not exists in obcluster here or add the validation logic in webhook